lib/python/rose/reporter.py

```diff
@@ -311,12 +308,14 @@ def write(self, message):
         self._send_pending_messages()
 
     def _send_pending_messages(self):
-        for message in list(self._messages_pending):
+        while self._messages_pending:
+            message = self._messages_pending[0]
```

Comment:
```python
message = self._messages_pending.pop()
```

Reply:
The lead message only gets `del` on successful queuing.